### PR TITLE
PoC download externally stored evaluator logs

### PR DESCRIPTION
**Pull request by esbena**
Try running the command `Download External Logs for Performance Comparison`...

This will build the following directory structure:

```
$ find 'User/globalStorage/github.vscode-codeql/compare-performance' -type f 
User/globalStorage/github.vscode-codeql/compare-performance/logs-of/artifacts/codeql-dca-runners/codeql-dca-worker_javascript/2181621080/codeql-evaluator-log.json
User/globalStorage/github.vscode-codeql/compare-performance/logs-of/artifacts/codeql-dca-runners/codeql-dca-worker_javascript/2181601861/codeql-evaluator-log.json
User/globalStorage/github.vscode-codeql/compare-performance/artifacts/codeql-dca-runners/codeql-dca-worker_javascript/2181621080/codeql-evaluator-log.json.tar.gz
User/globalStorage/github.vscode-codeql/compare-performance/artifacts/codeql-dca-runners/codeql-dca-worker_javascript/2181601861/codeql-evaluator-log.json.tar.gz
```

### CODE REVIEW

File: extensions/ql-vscode/src/compare-performance/compare-performance-view.ts [hunk 1]
```diff
@@ -12,16 +21,27 @@ import { showAndLogExceptionWithTelemetry } from "../common/logging";
...
+
+type ComparePerformanceCommands = {
+  "codeQL.compare-performance.downloadExternalLogs": () => Promise<void>;
+};
 
 export class ComparePerformanceView extends AbstractWebview<
   ToComparePerformanceViewMessage,
   FromComparePerformanceViewMessage
 > {
+  private workingDirectory;
```
Comment (asgerf) [inline extensions/ql-vscode/src/compare-performance/compare-performance-view.ts:42]:
```suggestion
  private workingDirectory: string;
```
Does TypeScript infer types of fields with an initializer now, or does this have type `any`?

File: extensions/ql-vscode/src/compare-performance/compare-performance-view.ts [hunk 2]
```diff
@@ -94,4 +118,287 @@ export class ComparePerformanceView extends AbstractWebview<
...
+              `Skipping download to existing '${artifactDiskId}'...`,
+            );
+          } else {
+            await ensureDir(downloadPath);
+            void extLogger.log(
+              `Downloading from ${artifactDiskId} (bytes: ${bytes}) ${downloadPath}...`,
+            );
+            await this.fetchAndUnzip(url, downloadPath, progress);
+          }
+          if (existsSync(logPath) && readdirSync(logPath).length >= 0) {
```
Comment (asgerf) [inline extensions/ql-vscode/src/compare-performance/compare-performance-view.ts:187]:
```suggestion
          if (existsSync(logPath) && readdirSync(logPath).length > 0) {
```
`.length >= 0` is always true.

File: extensions/ql-vscode/src/compare-performance/compare-performance-view.ts [hunk 3]
```diff
@@ -94,4 +118,287 @@ export class ComparePerformanceView extends AbstractWebview<
...
+    const url2 =
+      "https://github.com/codeql-dca-runners/codeql-dca-worker_javascript/actions/runs/11816721194/artifacts/2181601861";
+
+    await Promise.all([downloadLog(url1), downloadLog(url2)]);
+    void extLogger.log(`Downloaded logs to ${this.workingDirectory}`);
+
+    return;
+  }
+
+  private async fetchAndUnzip(
```
Comment (asgerf) [inline extensions/ql-vscode/src/compare-performance/compare-performance-view.ts:229]:
```suggestion
  private async fetchAndUnzip( // Copy-pasted from database-fetcher.ts
```

File: extensions/ql-vscode/src/compare-performance/compare-performance-view.ts (hunk 1 above)
Comment (esbena) [reply extensions/ql-vscode/src/compare-performance/compare-performance-view.ts:42]: it is typed as `string` everywhere 🎉

**Comment by esbena** [comment]
comments addressed (I'd like to keep the commit message)

**Pull request merged**